UIforETW/Utility.cpp

```diff
@@ -431,108 +645,117 @@ bool Is64BitWindows()
-		return kWindowsVersionVista;
+bool IsWindowsSevenOrLesser()
+{
+	return !IsWindows8OrGreater();
+}
 
-	return kWindowsVersionXP;
+bool IsWindowsVistaOrLesser()
+{
+	return !IsWindows7OrGreater();
 }
 
+std::wstring GetEnvironmentVariableString(_In_z_ PCWSTR const variable)
+{
+	const rsize_t bufferSize = 512u;
+	wchar_t buffer[bufferSize] = {0};
+	rsize_t sizeRequired = 0u;
+	const errno_t result = _wgetenv_s(&sizeRequired, buffer, variable);
+	if (result == 0)
+		return buffer;
+	return L"";
+}
 
+std::string GetEnvironmentVariableString(_In_z_ PCSTR const variable)
+{
+	const rsize_t bufferSize = 512u;
+	char buffer[bufferSize] = {0};
+	rsize_t sizeRequired = 0u;
+	const errno_t result = getenv_s(&sizeRequired, buffer, variable);
+	if (result == 0)
+		return buffer;
+	return "";
+}
 
 std::wstring FindPython()
 {
-#pragma warning(suppress:4996)
-	PCWSTR const pytwoseven = _wgetenv( L"python27" );
+	const std::wstring pytwoseven = GetEnvironmentVariableString( L"python27" );
 	
 	//Some people, like me, (Alexander Riccio) have an environment variable 
 	//that specifically points to Python 2.7.
 	//As a workaround for issue #13, we'll use that version of Python.
 	//See the issue: https://github.com/google/UIforETW/issues/13
-	if ( pytwoseven )
-	{
+	if ( !pytwoseven.empty() )
 		return pytwoseven;
+	const std::wstring path = GetEnvironmentVariableString(L"path");
+	if (path.empty())
+	{
+		// No python found.
+		return L"";
 	}
 
-#pragma warning(suppress:4996)	
-	const wchar_t* path = _wgetenv(L"path");
-	if (path)
+	const std::vector<std::wstring> pathParts = split(path, ';');
+	// First look for python.exe. If that isn't found then look for
+	// python.bat, part of Chromium's depot_tools
+	for (const auto& exeName : { L"\\python.exe", L"\\python.bat" })
 	{
-		std::vector<std::wstring> pathParts = split(path, ';');
-		// First look for python.exe. If that isn't found then look for
-		// python.bat, part of Chromium's depot_tools
-		for (const auto& exeName : { L"\\python.exe", L"\\python.bat" })
+		for (const auto& part : pathParts)
 		{
-			for (const auto& part : pathParts)
-			{
-				std::wstring pythonPath = part + exeName;
-				if (PathFileExists(pythonPath.c_str()))
-				{
-					return pythonPath;
-				}
-			}
+			const std::wstring pythonPath = part + exeName;
+			if (::PathFileExistsW(pythonPath.c_str()))
+				return pythonPath;
 		}
 	}
 	// No python found.
 	return L"";
 }
 
-std::wstring GetBuildTimeFromAddress(void* codeAddress)
+std::wstring GetBuildTimeFromAddress(_In_ const void* const codeAddress)
 {
 	// Get the base of the address reservation. This lets this
 	// function be passed any function or global variable address
 	// in a DLL or EXE.
-	MEMORY_BASIC_INFORMATION	memoryInfo;
-	if (VirtualQuery(codeAddress, &memoryInfo, sizeof(memoryInfo)) != sizeof(memoryInfo))
+
+	//https://msdn.microsoft.com/en-us/library/windows/desktop/aa366902.aspx
+	//If the function fails, the return value is zero.

```

Comment:
More specifically, the return value is the number of bytes copied. I want *all* requested bytes copied, which is why I compare against the size I passed in.